### PR TITLE
[IOTDB-5884] Throw Exception when alter template with duplicate measurement

### PR DESCRIPTION
**Pull request by MarcosZyk**
### Problem

When there's duplicate measurement in one template extension request, there's no exception returned but the request is executed successfully and the registered measurement is the first one.

### Cause

When extending schema template on configNode, the measurement duplication will be ignored and the first one takes effect.

### Solution

Add duplication check on dataNode before sending RPC request to configNode.
Add deduplication in template auto extension during data insertion with insert records or tablets or TsFile load.

### CODE REVIEW

File: server/src/main/java/org/apache/iotdb/db/metadata/template/alter/TemplateExtendInfo.java [hunk 1]
```diff
@@ -102,6 +104,24 @@ public void addMeasurement(
...
+      if (set.contains(measurements.get(i))) {
+        continue;
+      }
+      set.add(measurements.get(i));
+      result.addMeasurement(
+          measurements.get(i), dataTypes.get(i), encodings.get(i), compressors.get(i));
+    }
+    return result;
+  }
+
```
Comment (Cpaulyz) [inline server/src/main/java/org/apache/iotdb/db/metadata/template/alter/TemplateExtendInfo.java:124]:
```suggestion
  public boolean isDuplicated() {
    if (measurements != null) {
      Set<String> set = new HashSet<>(measurements);
      return set.size()!=measurements.size();
    }
    return false;
  }
```

File: server/src/main/java/org/apache/iotdb/db/mpp/plan/execution/config/executor/ClusterConfigTaskExecutor.java [hunk 2]
```diff
@@ -1382,6 +1387,28 @@ public SettableFuture<ConfigTaskResult> dropSchemaTemplate(
...
+          future.setException(
+              new MetadataException(
+                  String.format(
+                      "Duplicated measurement [%s] in schema template alter request",
+                      measurement)));
+          return future;
+        } else {
+          existingMeasurements.add(measurement);
+        }
+      }
```
Comment (Cpaulyz) [inline server/src/main/java/org/apache/iotdb/db/mpp/plan/execution/config/executor/ClusterConfigTaskExecutor.java:1409]:
```suggestion
      TemplateExtendInfo templateExtendInfo =
          (TemplateExtendInfo) alterSchemaTemplateStatement.getTemplateAlterInfo();
      if (templateExtendInfo.isDuplicated()) {
          future.setException(
              new MetadataException(
                  String.format(
                      "Duplicated measurement [%s] in schema template alter request",
                      measurement)));
          return future;
      }
```

**Review by Cpaulyz — COMMENTED** [review]
PTAL

File: server/src/main/java/org/apache/iotdb/db/mpp/plan/execution/config/executor/ClusterConfigTaskExecutor.java (hunk 2 above)
Comment (MarcosZyk) [reply server/src/main/java/org/apache/iotdb/db/mpp/plan/execution/config/executor/ClusterConfigTaskExecutor.java]: Already add ```TemplateExtendInfo#getFirstDuplicateMeasurement``` and use it.

File: server/src/main/java/org/apache/iotdb/db/metadata/template/alter/TemplateExtendInfo.java (hunk 1 above)
Comment (MarcosZyk) [reply server/src/main/java/org/apache/iotdb/db/metadata/template/alter/TemplateExtendInfo.java:139]: Already add ```TemplateExtendInfo#getFirstDuplicateMeasurement``` and use it.

**Review by Cpaulyz — APPROVED** [review]
LGTM

**Pull request merged**